internal/download/get.go

```diff
@@ -39,66 +40,80 @@ func SetMocking(useMocking bool) {
 	}
 }
 
-func httpGet(url string) ([]byte, *failures.Failure) {
+func httpGet(url string) ([]byte, error) {
 	logging.Debug("Retrieving url: %s", url)
 	return httpGetWithProgress(url, nil)
 }
 
-func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, *failures.Failure) {
-	logging.Debug("Retrieving url: %s", url)
-	client := retryhttp.NewClient(0 /* 0 = no timeout */, 5)
-	resp, err := client.Get(url)
-	if err != nil {
-		code := -1
-		if resp != nil {
-			code = resp.StatusCode
+func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, error) {
+	var bs []byte
+	fn := func() error {
+		logging.Debug("Retrieving url: %s", url)
+		client := retryhttp.NewClientFromExisting(retryhttp.DefaultClient, 3)
+		resp, err := client.Get(url)
+		if err != nil {
+			code := -1
+			if resp != nil {
+				code = resp.StatusCode
+			}
+			fail := failures.FailNetwork.Wrap(err, locale.Tl("err_network_get", "Status code: {{.V0}}", strconv.Itoa(code)))
+			return &retryfn.ControlError{
+				Cause: fail,
+				Type:  retryfn.Halt,
+			}
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != 200 {
+			fail := failures.FailNetwork.New("err_invalid_status_code", strconv.Itoa(resp.StatusCode))
+			return &retryfn.ControlError{
+				Cause: fail,
+				Type:  retryfn.Halt,
```

Comment:
Wouldn't this cause it NOT to retry?

Reply:
In the case of "statusCode != 200":
IIUC, if the flow reaches here, the server has responded with some kind of affirmative and retrying is unlikely to then produce a 200.

In the case of "client.Get()-err != nil":
I was trying to prevent trying so many times, but I think it makes sense now that we're setting max tries to 3x3, so that is updated.